### ISSUE TITLE
mismatch for inception3a.branch3.1.conv.weight: copying a param with shape torch.Size([32, 16, 3, 3]) from checkpoint, the shape in current model is torch.Size([32, 16, 5, 5]).

### ISSUE THREAD

**Issue by wang66624** [WZMIAOMIAO/deep-learning-for-image-processing#539]
https://github.com/WZMIAOMIAO/deep-learning-for-image-processing/blob/87a31d3db4baa5a693c59cab66cb232357c326c9/pytorch_classification/Test4_googlenet/train.py#L65
您好，我加载了googlenet的预训练权重会出现标题所指问题，请问如何解决？如果我修改branch3的kernal size为3 则则会出现RuntimeError: Sizes of tensors must match except in dimension 2. Got 28 and 30 (The offending index is 2)

**Comment by WZMIAOMIAO** [comment]
载入预训练权重需要把和类别相关的权重全部删掉，下面这几行代码都是需要使用的，不是仅仅使用一个torch.load的

https://github.com/WZMIAOMIAO/deep-learning-for-image-processing/blob/87a31d3db4baa5a693c59cab66cb232357c326c9/pytorch_classification/Test4_googlenet/train.py#L63-L71

**Comment by wang66624** [comment]
嗯嗯，我都加了，但是我发现您的网络参数跟原GoogLeNet不一样，
 (branch3): Sequential(
      (0): BasicConv2d(
        (conv): Conv2d(512, 24, kernel_size=(1, 1), stride=(1, 1))
        (relu): ReLU(inplace=True)
      )
      (1): BasicConv2d(
        (conv): Conv2d(24, 64, kernel_size=(5, 5), stride=(1, 1), padding=(2, 2))
        (relu): ReLU(inplace=True)
      )
    )
我加载的预训练网络branch3中下面那个kernal_size=(3,3),padding=(1,1)
改成这个了以后才不会报mismatch的错

**Comment by wang66624** [comment]
![image](https://user-images.githubusercontent.com/81811919/165473284-7357cb0a-c94f-4413-848d-58f1548c2f65.png)
用您这段代码会出现这样的错误，网络结构对应不上

**Comment by WZMIAOMIAO** [comment]
我实现的模型和pytorch官方给的确实不一样，关于你说的问题可以看下这个issue。我提供的代码是将预训练权重直接载入官方模型使用不是载入我提供的模型，不是载入我提供的模型，不是载入我提供的模型。正确的的使用发下如下：
```python
net = torchvision.models.googlenet(num_classes=5)
model_dict = net.state_dict()
pretrain_model = torch.load("googlenet.pth")
del_list = ["aux1.fc2.weight", "aux1.fc2.bias",
            "aux2.fc2.weight", "aux2.fc2.bias",
            "fc.weight", "fc.bias"]
pretrain_dict = {k: v for k, v in pretrain_model.items() if k not in del_list}
model_dict.update(pretrain_dict)
net.load_state_dict(model_dict)
# net = GoogLeNet(num_classes=5, aux_logits=True, init_weights=True)
# net.to(device)
```

**Comment by wang66624** [comment]
好的谢谢~如果要自己在网络中添加模块是不是就不能用官方模型了，得直接自己创建模型并在其中加入模块？

**Comment by WZMIAOMIAO** [comment]
> 好的谢谢~如果要自己在网络中添加模块是不是就不能用官方模型了，得直接自己创建模型并在其中加入模块？

是的，需要在自己的模型中就行修改，不要去动源码。

**Comment by wang66624** [comment]
感谢，还想再问一个问题>_<,我手头的数据很少，想要进行数据扩充是对训练集扩充n倍以后再从中划分训练集验证集吗？这样验证集和训练集中是不是不能包含同一张图经扩充后的图像？还是无所谓啊随机划分就行？（两百张不到，5类，图像标注缓慢又艰难，大概多深的网络能不过拟合嘞==）

**Comment by xiongzhu666** [comment]
> 感谢，还想再问一个问题>_<,我手头的数据很少，想要进行数据扩充是对训练集扩充n倍以后再从中划分训练集验证集吗？这样验证集和训练集中是不是不能包含同一张图经扩充后的图像？还是无所谓啊随机划分就行？（两百张不到，5类，图像标注缓慢又艰难，大概多深的网络能不过拟合嘞==）

我现在也是这种情况，4个类，每个类5百张，严重过拟合....gridmask的图像增强都用了，还是没用
